juice/src/net/activation/sigmoid.rs

```diff
@@ -0,0 +1,72 @@
+use crate::co::IBackend;
+use crate::conn;
+use crate::net::{Context, Descriptor, Layer};
+
+#[derive(Debug, Clone)]
+pub struct Sigmoid {
+    descriptor: Descriptor,
+}
+
+impl Sigmoid {
+    pub fn new(mut descriptor: Descriptor) -> Self {
+        assert_eq!(descriptor.inputs().len(), 1); // Should only be one input.
+
+        descriptor.add_output(descriptor.input(0).unit_shape().clone());
+
+        Sigmoid {
+            descriptor: descriptor,
+        }
+    }
+}
+
+impl<B: IBackend + conn::Sigmoid<f32>> Layer<B> for Sigmoid {
+    fn compute_output(&self, backend: &B, context: &mut Context) {
+        let input = context.get_data(self.descriptor.input(0));
+        let output = context.acquire_data(self.descriptor.output(0));
+        backend
+            .sigmoid(&input.borrow(), &mut output.borrow_mut())
+            .unwrap();
```

Comment:
Note to self: Assure the `unwrap()` is never hit, or alternatively make it return a `Result` with a really small error type, and bench the perf change.

Reply:
Ack. (Resolving.)